src/OSPSuite.Infrastructure.Import/Core/DataSourceFileReaders/ExcelReader.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using NPOI.HSSF.UserModel;
+using NPOI.SS.UserModel;
+using NPOI.XSSF.UserModel;
+
+namespace OSPSuite.Infrastructure.Import.Core.DataSourceFileReaders
+{
+   public class ExcelReader //or maybe rename it to excelReaderExtensions - or make it into extensions
+   {
+      private const string _xlsxExtension = ".xlsx";
+      //we could get this info here or in getCurrentRow, but make sure we do it once
+      private IWorkbook _book;
+      private IEnumerator<ISheet> _sheetEnumerator;
+      private IEnumerator _rowEnumerator;
+      private bool _columnOffsetOn;
+      private int _columnOffset;
+      private bool _isTypeXlsx;
+      private IFormulaEvaluator _formulaEvaluator;
+
+      public ISheet CurrentSheet { get; set; } //why is this public even????
+      public List<string> CurrentRow { get; set; } = new List<string>();
```

Comment:
I changed the setters to be private - since the setting is only being done within the class. CurrentSheet is public because the name of the sheet needs to be read from the client - I have removed the comment.